jbpm-flow/src/test/java/org/jbpm/process/core/impl/ObjectClonerTest.java

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.process.core.impl;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertSame;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.jbpm.process.core.impl.ObjectCloner.Config;
+import org.junit.Test;
+
+public class ObjectClonerTest {
```

Comment:
you could add also some tests for rainy scenarios and exceptions like ReflectiveOperationException, wdyt?

Reply:
hmmm, I do no think is a case where this exception is going to be throwb, since this is reflection inside the same object and Im using public methods (Im catching because the compiles enforces me to do so)